cfme/scripting/polarion.py

```diff
@@ -0,0 +1,131 @@
+import click
+import ruamel.yaml
+from miq_version import Version
+from ruamel.yaml.util import load_yaml_guess_indent
+
+from cfme.utils.conf import credentials
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+from cfme.utils.path import conf_path
+
+
+# constants for the polarion fields in yaml
+POLARION_PROJECT_ID = 'polarion-project-id'
+POLARION_URL = 'polarion_url'
+XUNIT_HEADER = 'xunit_import_properties'
+# these are under xunit import
+TEMPLATE_ID = 'polarion-testrun-template-id'
+TESTRUN_TITLE = 'polarion-testrun-title'
+TESTRUN_ID = 'polarion-testrun-id'
+GROUP_ID = 'polarion-group-id'
+
+# Log to stdout too
+add_stdout_handler(logger)
+
+
+@click.group()
+def main():
+    """Function to populate polarion_tools.local.yaml
+
+    This tool will take a polarion_tools.local.yaml template file, and populate empty field values
+    with those derived from the given CLI options
+
+    The default behavior is to write to polarion_tools.local.yaml in the conf_path directory
+    cfme_testcases_upload will combine local.yaml and polarion_tools.yaml
+    """
+    pass
+
+
+@main.command('populate',
+              help='Populate template with credentials and version based testrun data.'
+              'Does NOT overwrite anything already set in the given template')
+@click.option('--template',
+              help='YAML template to start with',
+              default=conf_path.join('polarion_tools.local.yaml.template').strpath)
+@click.option('--output',
+              help='YAML file to write to',
+              default=conf_path.join('polarion_tools.local.yaml').strpath)
+@click.option('--credentials_key',
+              help='Key for utils.conf.credentials used for polarion authentication',
+              default='polarion-upload')
+@click.option('--vstring',
+              help='version string that can be parsed for series and template selection')
+@click.option('--polarion-project-id',
+              help='The polarion project ID to inject',
+              default=None)
+@click.option('--polarion-url',
+              help='The polarion URL to inject',
+              default=None)
+@click.option('--template-format',
+              help='The format for the test template, y-stream CFME digits is formatted in',
+              default='ImportTestRunTemplate')
+def populate(
```

Comment:
do you like such formatting ?

Reply:
Kinda, not so much, I dropped them to a newline. Its quite verbose in general trying to decorate `click` methods with these.

Feel free to format as you like, and I'll merge the PR :)